BeeSwift/GoalViewModel.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  GoalViewModel.swift
+//  BeeSwift
+//
+//  Created by krugerk on 2024-11-25.
+//
+
+import Foundation
+import Intents
+
+import BeeKit
+
+struct GoalViewModel {
```

Comment:
I like having somewhere for view-specific logic for goals to live which isn't in the model, and isn't in the UI classes. I don't think we've reached a point where it makes sense for that to be the main object we pass around though. 
1. It adds unnecessary indirection
2. We might end up wanting different "view models" for different views.

Reply:
There is one approach where each and every view has its own view model.